snuba/datasets/schemas/__init__.py

```diff
@@ -0,0 +1,68 @@
+from abc import ABC, abstractmethod
+from typing import Mapping, List
+
+from snuba import settings
+from snuba.clickhouse.columns import ColumnSet
+
+
+def local_dataset_mode():
+    return settings.DATASET_MODE == "local"
+
+
+class Schema(ABC):
+    """
+    A schema is an abstraction over a data model we can query.
+    It provides a set of columns and a where clause to build the query.
+    Concretely this can represent a table, a view or a group of
+    joined tables.
+    This level of abstraction only provides read primitives.
+
+    As of now this we do not have a strict separation between a Snuba abstract
+    schema and a Clickhouse concrete schema. When this will exist, this
+    class will break up into snuba schema and clickhouse schema.
+    """
+
+    def __init__(
+        self,
+        *,
+        columns: ColumnSet,
```

Comment:
I think it'd make sense to make columns able to be passed positionally, especially since that kind of implies that all subclasses should be accepting that as an argument, which seems like a safe assumption?